autopush/settings.py

```diff
@@ -359,13 +383,23 @@ def parse_endpoint(self, token, version="v1", ckey_header=None,
         """
         token = self.fernet.decrypt(repad(token).encode('utf8'))
         public_key = None
+        self.vapid_auth = None
         if ckey_header:
             try:
                 crypto_key = CryptoKey(ckey_header)
             except CryptoKeyException:
                 raise InvalidTokenException("Invalid key data")
             public_key = crypto_key.get_label('p256ecdsa')
-
+        if auth_header:
+            self.vapid_auth = self.extract_auth(auth_header)
```

Comment:
Um, so we're setting a global settings object's vapid_auth (used by multiple threads) to a request specific object?